client/src/components/FriendRequest.tsx

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from 'react';
+import { FaUserFriends } from 'react-icons/fa';
+import { AiOutlineCheck } from 'react-icons/ai';
+import {
+  useGetUserFriendRequestsQuery,
+  useAcceptFriendRequestMutation,
+  useRejectFriendRequestMutation,
+} from '../redux/services/user/userService';
+import { User as UserType } from '../types';
+import { Spinner } from 'flowbite-react';
+import { Link } from 'react-router-dom';
+function FriendRequest() {
+  const { data, isLoading } = useGetUserFriendRequestsQuery('User');
+  const [friendRequests, setFriendRequests] = useState<UserType[]>();
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [acceptFriendRequest] = useAcceptFriendRequestMutation();
+  const [rejectFriendRequest] = useRejectFriendRequestMutation();
+  useEffect(() => {
+    if (data && !isLoading) {
+      setFriendRequests(data);
+    }
+  }, [data, isLoading]);
+
+  const handleBellClick = () => {
+    setIsDropdownOpen((prevState) => !prevState);
+  };
+
+  const handleIconClick = (event: any) => {
+    event.stopPropagation();
+    setIsDropdownOpen((prevState) => !prevState);
+  };
+
+  const acceptFriendRequestHandler = async (id: string) => {
```

Comment:
Question ❓ - Is there a reason why we gotta wrap the `acceptFriendRequest` and the `rejectFriendRequest` functions inside of the handler functions instead of directly using them in the button's `onClick` attribute?

Reply:
![image](https://github.com/KyoshiNoda/ScheduleFinder/assets/62672803/3898fc98-b0aa-419e-bc18-76d1aff812ee)

Reply:
It seems that typescript is mad that its a promised based for `onClick` so to get by this we pass it as a callback function where we wrap it

Reply:
<img width="716" alt="Screenshot 2023-08-21 at 10 38 00 PM" src="https://github.com/KyoshiNoda/ScheduleFinder/assets/62437724/9317df51-7e2b-42cf-a66c-f832d7c46f7b">
Tried this and it doesn't seem to complain, which is weird. It's ok tho, not big deal.